backend/src/serverless/integrations/services/integrations/hackerNewsIntegrationService.ts

```diff
@@ -0,0 +1,179 @@
+  constructor() {
+    super(IntegrationType.HACKER_NEWS, 2 * 60)
+  }
+
+  async createMemberAttributes(context: IStepContext): Promise<void> {
+    const service = new MemberAttributeSettingsService(context.serviceContext)
+
+    await service.createPredefined(HackerNewsMemberAttributes)
+  }
+
+  async preprocess(context: IStepContext): Promise<void> {
+    const settings = context.integration.settings as HackerNewsIntegrationSettings
+
+    const keywords = Array.from(new Set([...settings.keywords, ...settings.urls]))
+
+    const posts = await getPostsByKeywords(
+      { keywords, nDays: context.onboarding ? 1000000 : 3 },
+      context.serviceContext,
+      this.logger(context),
+    )
+
+    context.pipelineData = {
+      keywords,
+      posts,
+    }
+  }
+
+  async getStreams(context: IStepContext): Promise<IIntegrationStream[]> {
+    return context.pipelineData.posts.map((a: EagleEyeResponse) => ({
+      value: a.sourceId.slice(a.sourceId.lastIndexOf(':') + 1),
+      metadata: {
+        channel: a.keywords[0],
+      },
+    }))
+  }
+
+  async processStream(
+    stream: IIntegrationStream,
+    context: IStepContext,
+  ): Promise<IProcessStreamResults> {
+    const logger = this.logger(context)
+    let newStreams: IIntegrationStream[]
+
+    const post: HackerNewsResponse = await getPost(stream.value, logger)
+
+    if (post.kids !== undefined) {
+      newStreams = post.kids.map((a: number) => ({
+        value: a.toString(),
+        metadata: {
+          ...stream.metadata,
+          ...((!post.parent && {
+            parentId: post.id.toString(),
+            parentTitle: post.title || post.text,
+          }) ||
+            {}),
+        },
+      }))
+    }
+
+    let activities: AddActivitiesSingle[]
+    if (!post.text && !post.url) {
+      activities = []
+    } else {
+      const parsedPost = this.parsePost(
+        context.integration.tenantId,
+        stream.metadata.channel,
+        post,
+        stream.metadata.parentId,
+        stream.metadata.parentTitle,
+      )
+      activities = [parsedPost]
+    }
+
+    const lastRecord = activities.length > 0 ? activities[activities.length - 1] : undefined
+
+    let sleep: number | undefined
+
+    return {
+      operations: [
+        {
+          type: Operations.UPSERT_ACTIVITIES_WITH_MEMBERS,
+          records: activities,
+        },
+      ],
+      lastRecord,
+      lastRecordTimestamp: lastRecord ? lastRecord.timestamp.getTime() : undefined,
+      sleep,
+      newStreams,
+    }
+  }
+
+  parsePost(
+    tenantId,
+    channel,
+    post: HackerNewsResponse,
+    parentId,
+    parentTitle,
+  ): AddActivitiesSingle {
+    const type = post.parent ? 'comment' : 'post'
+    const url = `https://news.ycombinator.com/item?id=${post.parent ? post.parent : post.id}`
+    const body =
+      post.text !== undefined && post.text !== ''
+        ? sanitizeHtml(post.text)
+        : `<a href="${post.url}" target="_blank">${post.url}</a>`
```

Comment:
No, that would not work. HackerNews will either give us a body (HTML) or a URL where the post points to. We are setting this URL as the body, in case there is no body. The frontend cannot know which option is where it comes from.